base/compiler/ssair/passes.jl

```diff
@@ -1913,8 +1913,6 @@ function perform_symbolic_evaluation(stmt::PhiNode, ssa_to_ssa, blockidx)
     key[no_of_edges+2] = blockidx
     copyto!(key, no_of_edges+3, stmt.values)
 
-    firstval = nothing
-    allthesame = true # If all values into phi node are the same SSAValue
```

Comment:
A missing edge is `undef` in the LLVM sense, so any value is legal. The issue is that the value may not be available on that predecessor, so you need to check dominance.